### PR TITLE
std: move allocators to `sys`

### PR DESCRIPTION
**Pull request by joboet**
Part of #117276.

### CODE REVIEW

File: library/std/src/sys/pal/teeos/alloc.rs
Comment (joboet) [inline library/std/src/sys/pal/teeos/alloc.rs:1]: This was identical to the UNIX version.

File: src/tools/miri/tests/fail/alloc/global_system_mixup.stderr [hunk 1]
```diff
@@ -1,13 +1,13 @@
 error: Undefined Behavior: deallocating ALLOC, which is Rust heap memory, using PLATFORM heap deallocation operation
-  --> RUSTLIB/std/src/sys/pal/PLATFORM/alloc.rs:LL:CC
+  --> RUSTLIB/std/src/sys/alloc/unix.rs:LL:CC
```
Comment (RalfJung) [inline src/tools/miri/tests/fail/alloc/global_system_mixup.stderr:2]: This won't work, the test is also run on Windows targets.

The test will need a `normalize-stderr-test` that normalizes `alloc/$OS` to always produce the same output.

**Comment by cuviper** [comment]
LGTM!

@bors r+ iffy

**Comment by joboet** [comment]
I forgot to check SGX and forgot some imports...

@bors r=cupiver

**Comment by joboet** [comment]
That looks spurious.
@bors retry

**Pull request merged**